test/sanity/validate-modules/main.py

```diff
@@ -1229,11 +1229,26 @@ def _validate_argument_spec(self, docs, spec, kwargs):
 
             # TODO: needs to recursively traverse suboptions
             doc_type = docs.get('options', {}).get(arg, {}).get('type', 'str')
-            if 'type' in data and data['type'] == 'bool' and doc_type != 'bool':
+            if 'type' in data:
+                if data['type'] == 'bool' and doc_type != 'bool':
+                    self.reporter.error(
+                        path=self.object_path,
+                        code=325,
+                        msg='argument_spec for "%s" defines type="bool" but documentation does not' % (arg,)
+                    )
+
+                if data['type'] != doc_type:
+                    self.reporter.error(
+                        path=self.object_path,
+                        code=336,
+                        msg='argument_spec for "%s" defines type as %s but documentation defines type as %s' % (arg, data['type'], doc_type)
+                    )
+
+            if 'type' not in data and doc_type in frozenset(('list', 'dict', 'bool', 'int', 'float')):
```

Comment:
Shouldn't there also be `path` and some more in this list?

Reply:
That list should be shared with the Schema tests :)

Reply:
But to be honest, the types should be identical, it shouldn't matter what it is.